sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java

```diff
@@ -207,26 +271,47 @@ PubsubIO.Read<PubsubMessage> buildPubsubRead() {
         throw new IllegalArgumentException(
             "Both PubsubTestClientFactory and Clock need to be specified for testing, but only one is provided");
       }
+      if (!Strings.isNullOrEmpty(configuration.getIdAttribute())) {
+        pubsubRead = pubsubRead.withIdAttribute(configuration.getIdAttribute());
+      }
+      if (!Strings.isNullOrEmpty(configuration.getTimestampAttribute())) {
+        pubsubRead = pubsubRead.withTimestampAttribute(configuration.getTimestampAttribute());
+      }
       return pubsubRead;
     }
 
     @Override
     public PCollectionRowTuple expand(PCollectionRowTuple input) {
       PubsubIO.Read<PubsubMessage> pubsubRead = buildPubsubRead();
+      @SuppressWarnings("nullness")
```

Comment:
Is this suppression needed ?(if so probably can be removed by adding a null check before this line).

Reply:
Unfortunately the nullness checker is not smart enough for that.